crates/oxc_linter/src/rules/eslint/no_case_declarations.rs

```diff
@@ -65,9 +65,10 @@ impl Rule for NoCaseDeclarations {
                     Statement::VariableDeclaration(var) if var.kind.is_lexical() => {
                         let start = var.span.start;
                         let end = match var.kind {
-                            VariableDeclarationKind::Const => 5,
+                            VariableDeclarationKind::Const | VariableDeclarationKind::Using => 5,
                             VariableDeclarationKind::Let => 3,
-                            _ => unreachable!(),
+                            VariableDeclarationKind::AwaitUsing => var.span.size(),
```

Comment:
This line doesn't look quite right. `var.span.size()` will be the length of the entire `VariableDeclaration`, whereas the other arms are just returning the length of the `const` / `let` / `using` keyword.

Maybe that doesn't matter!

But if it does, something like `source_text[var.span.start as usize..var.declarations[0].span.start as usize].trim_end().len() as u32` might do it? Pretty ugly!

Reply:
i added a test for this case. the problem with `await using` is that it could be any length, and trying to workout the exact span probably isn't work it right now.

Instead i changed it's span to be from the start of the `await using` to the start of the span of the first var decl. this should guarentee that. the span is in valid unicode, while still pointing to `await using` keywords in most cases.